src/pevm.rs

```diff
@@ -328,6 +262,54 @@ pub fn execute_revm_sequential<S: Storage>(
     Ok(results)
 }
 
+fn canonize_account(
+    results: &mut [PevmTxExecutionResult],
+    address: Address,
+    initial_account_basic: AccountBasic,
+    write_history: BTreeMap<TxIdx, MemoryEntry>,
+) {
+    let mut current_account = initial_account_basic;
+
+    for (tx_idx, memory_entry) in write_history {
+        match memory_entry {
+            MemoryEntry::Data(_, MemoryValue::Basic(info)) => {
+                current_account.balance = info.balance;
+                current_account.nonce = info.nonce;
+            }
+            MemoryEntry::Data(_, MemoryValue::LazyBalanceAddition(addition)) => {
+                current_account.balance += addition;
+            }
+            _ => unreachable!(),
+        }
+
+        let result = unsafe { results.get_unchecked_mut(tx_idx) };
+        match result.state.entry(address) {
+            std::collections::hash_map::Entry::Occupied(occupied) => {
+                let account = occupied.into_mut();
+                if current_account.is_empty() {
+                    *account = None;
+                } else if let Some(account) = account {
+                    account.basic.balance = current_account.balance;
+                    account.basic.nonce = current_account.nonce;
+                } else {
+                    // for beneficiary only
```

Comment:
This will be for reward addresses like OP's as well, right? Let's do 

```
For implicit rewards like the beneficiary account that doesn't have explicit writes in [result.state]
```